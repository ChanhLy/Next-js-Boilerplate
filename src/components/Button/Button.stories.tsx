import React from 'react';

import { Meta, Story } from '@storybook/react';

import { Button, ButtonProps } from './Button';

const story: Meta = {
  title: 'Components/Button',
  component: Button,
};

export const Default: Story = (props: ButtonProps) => {
  return <Button {...props}>Primary</Button>;
};

Default.args = {
  children: 'text-red-300',
};
Default.parameters = {
  jest: ['Button.test.tsx'],
};

export default story;
