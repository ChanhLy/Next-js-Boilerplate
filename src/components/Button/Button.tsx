export interface ButtonProps {
  className?: string;
  onClick?: () => void;
}

export const Button: React.FC<ButtonProps> = (props) => {
  return (
    <button type="button" {...props}>
      {props.children}
    </button>
  );
};
