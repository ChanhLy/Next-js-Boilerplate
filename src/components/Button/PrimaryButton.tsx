import React from 'react';

import { Button, ButtonProps } from './Button';

export function PrimaryButton(props: ButtonProps) {
  return (
    <Button className="bg-blue-400 text-white" {...props}>
      Primary
    </Button>
  );
}
