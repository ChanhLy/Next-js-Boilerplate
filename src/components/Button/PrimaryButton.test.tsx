import React from 'react';

import renderer from 'react-test-renderer';

import { PrimaryButton } from '@/components/Button/PrimaryButton';

it('renders correctly', () => {
  const tree = renderer.create(<PrimaryButton>Button</PrimaryButton>).toJSON();
  expect(tree).toMatchSnapshot();
});
