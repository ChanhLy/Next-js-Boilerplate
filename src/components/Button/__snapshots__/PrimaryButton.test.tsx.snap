// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="bg-blue-400 text-white"
  type="button"
>
  Primary
</button>
`;
